kernel/comps/framebuffer/src/ansi_escape.rs

```diff
@@ -161,17 +161,75 @@ impl EscapeFsm {
                 }
 
                 // Set foreground and background colors
-                // Reference: <https://en.wikipedia.org/wiki/ANSI_escape_code#3-bit_and_4-bit>
+                // Reference: <https://en.wikipedia.org/wiki/ANSI_escape_code#Colors>
                 30..=37 => op.set_fg_color(COLORS[*param as usize - 30]),
+                38 => {
+                    if num_params == 3 && self.params[1] == 5 {
+                        op.set_fg_color(Self::get_color(self.params[2] as u8));
+                        return;
```

Comment:
This works for `[38;2;41;41;41`, but I don't believe this work for `[41;38;2;41;41;41` or `[38;2;41;41;41;41`. We should convert the `for` loop to a `while` loop to be able to support them? Also there can be some tests for such tricky things.